### PR TITLE
Add Support for Creality 3D Printer Models K1 and K1 Max

### PR DESCRIPTION
**Pull request by iZonex**
## Add Support for Creality 3D Printer Models K1 and K1 Max

Screenshot 2023-11-18 at 00 49 47
Screenshot 2023-11-18 at 00 49 30
Screenshot 2023-11-18 at 00 49 47
Screenshot 2023-11-18 at 14 09 05

### Introduction
This pull request integrates support for two new Creality 3D printer models: **K1** and **K1 Max** into Bambu Studio for software, enhancing our range of compatible devices.

#### Core Updates
- **Cover Images:** `Creality K1_cover.png`, `Creality K1 Max_cover.png`
- **Buildplate Models and Textures:** 
  - `creality_k1_buildplate_model.stl`
  - `creality_k1max_buildplate_model.stl`
  - `creality_k1_buildplate_texture.png`
  - `creality_k1max_buildplate_texture.png`
- **Machine-Specific Profiles:** 
  - `Creality K1 0.4 nozzle.json`
  - `Creality K1 Max 0.4 nozzle.json`
  - `Creality K1 Max.json`
  - `Creality K1.json`
- **Process Profiles:** 
  - `0.16mm Optimal @Creality 0.4 K1.json`
  - `0.20mm Standard @Creality 0.4 K1Max.json`
  - _and others for different print qualities_

#### Enhanced Compatibility
- `resources/profiles/Creality.json`
- `resources/profiles/Creality/filament/Generic ABS @Creality.json`
- `resources/profiles/Creality/filament/Generic PETG @Creality.json`
- `resources/profiles/Creality/filament/Generic PLA @Creality.json`
- `src/slic3r/GUI/CreatePresetsDialog.cpp`

### Rationale
Incorporating the **K1** and **K1 Max** models aligns with our goal of keeping our software current and accommodating a broader range of 3D printing technologies.

### Testing
Comprehensive testing ensures these profiles provide the best experience for K1 and K1 Max users.

### CODE REVIEW

**Comment by iZonex** [comment]
@lanewei120 We can add this to 1.8? I already tested on my K1/K1 Max printer, working ok, only one problem with thumbnails what was fixed in Orca Fork, but this issue not related to profile for K1/K1 Max

**Comment by lanewei120** [comment]
@SaltWei @XunZhangBambu please review

File: resources/profiles/Creality.json [hunk 1]
```diff
@@ -182,10 +198,26 @@
             "name": "0.20mm Standard @Creality Ender6",
             "sub_path": "process/0.20mm Standard @Creality Ender6.json"
         },
+        {
+			"name": "0.20mm Standard @Creality 0.4 K1",
+			"sub_path": "process/0.20mm Standard @Creality 0.4 K1.json"
+		},
+		{
+			"name": "0.20mm Standard @Creality 0.4 K1Max",
```
Comment (XunZhangBambu) [inline resources/profiles/Creality.json:206]: I prefer to name it as "xxx @Creality K1Max 0.4 nozzle"

File: resources/profiles/Creality/machine/Creality K1 Max 0.4 nozzle.json [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+		"40"
+	],
+	"z_hop": [
+		"0.4"
+	],
+	"single_extruder_multi_material": "1",
+	"change_filament_gcode": "",
+	"machine_pause_gcode": "PAUSE",
+	"default_filament_profile": [
+		"Creality Generic PLA"
```
Comment (XunZhangBambu) [inline resources/profiles/Creality/machine/Creality K1 Max 0.4 nozzle.json:119]: should be Generic PLA @Creality

File: resources/profiles/Creality/machine/Creality K1 0.4 nozzle.json [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+		"40"
+	],
+	"z_hop": [
+		"0.4"
+	],
+	"single_extruder_multi_material": "1",
+	"change_filament_gcode": "",
+	"machine_pause_gcode": "PAUSE",
+	"default_filament_profile": [
+		"Creality Generic PLA"
```
Comment (XunZhangBambu) [inline resources/profiles/Creality/machine/Creality K1 0.4 nozzle.json:119]: should be Generic PLA @Creality

**Review by XunZhangBambu — CHANGES_REQUESTED** [review]
After applying these commits, I also found that the memory usage becomes very high when switching the printer to K1 and K1 Max.

File: resources/profiles/Creality.json (hunk 1 above)
Comment (iZonex) [reply resources/profiles/Creality.json]: Thanks! Was renamed by yours pattern

File: resources/profiles/Creality/machine/Creality K1 Max 0.4 nozzle.json (hunk 2 above)
Comment (iZonex) [reply resources/profiles/Creality/machine/Creality K1 Max 0.4 nozzle.json]: Oh, Thanks! You found! Fixed

**Comment by iZonex** [comment]
> After applying these commits, I also found that the memory usage becomes very high when switching the printer to K1 and K1 Max.

Optimized sized stl and png files for plate and texture K1, K1 Max

**Comment by iZonex** [comment]
@XunZhangBambu Now all ok?

**Comment by XunZhangBambu** [comment]
Sorry,I still found  the memory usage becomes extremely high
20231205100939_rec_

**Comment by iZonex** [comment]
> @iZonex Sorry,I still find the memory usage becomes extremely high 20231205100939_rec_ [ 20231205100939_rec_ ](https://private-user-images.githubusercontent.com/139755545/287894687-28783494-d7b6-413f-884d-eb7fad300491.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3NjkyNDUsIm5iZiI6MTcwMTc2ODk0NSwicGF0aCI6Ii8xMzk3NTU1NDUvMjg3ODk0Njg3LTI4NzgzNDk0LWQ3YjYtNDEzZi04ODRkLWViN2ZhZDMwMDQ5MS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIwNVQwOTM1NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kY2M0YTZjZGE5YjhlYTkwNDhlN2YxY2RlYTk3ZTIxM2I3MDYzYzM4MDMyNTg0NTlhOWEzMTQzYTkzOTdjMjI2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.rtRnSKP4xXomL8W05oRUVR9t9rA1C8l9TM_Pu8IEzf8) [ ](https://private-user-images.githubusercontent.com/139755545/287894687-28783494-d7b6-413f-884d-eb7fad300491.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3NjkyNDUsIm5iZiI6MTcwMTc2ODk0NSwicGF0aCI6Ii8xMzk3NTU1NDUvMjg3ODk0Njg3LTI4NzgzNDk0LWQ3YjYtNDEzZi04ODRkLWViN2ZhZDMwMDQ5MS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIwNVQwOTM1NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kY2M0YTZjZGE5YjhlYTkwNDhlN2YxY2RlYTk3ZTIxM2I3MDYzYzM4MDMyNTg0NTlhOWEzMTQzYTkzOTdjMjI2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.rtRnSKP4xXomL8W05oRUVR9t9rA1C8l9TM_Pu8IEzf8)

Now I used simple version of build plates working much faster.

**Comment by iZonex** [comment]
https://github.com/bambulab/BambuStudio/assets/2759749/18e70aa6-5505-4c30-9566-836e3ae99c5c

**Comment by XunZhangBambu** [comment]
@iZonex  I tried again, but encountered an error while loading the STL. And I believe that just a STL file shouldn't result in such high memory usage

**Comment by XunZhangBambu** [comment]
@iZonex  After testing again ,that's my problem. Now I think this pull request can be merged. Excellent work ! Thanks very much for your efforts!

**Pull request merged**